application/common/models/Email.php

```diff
@@ -55,10 +60,10 @@ public function behaviors()
     /**
      * Attempt to send email. Returns true on success or throws exception.
      * DOES NOT QUEUE ON FAILURE
-     * @return void
+     * @return bool
      * @throws \Exception
      */
-    public function send()
+    public function send(): bool
```

Comment:
Returning a boolean from a function intended to do something (like `send()` is), rather than answer something (e.g. `isValid()`), seems to make code less readable in general. Even when dealing with code that does do something and return a boolean, such as Yii's models' `save()` methods, a false returned more often means that there was an error or failure, not merely that it was delayed.

Is there some other simple way to accomplish the goal here? If not, this seems like it will work. I just wanted to push back slightly due to that little code smell.

Reply:
One option would be to throw a custom exception type, which would be easy to use from `EmailController->actionQueue()` since it is already catching exceptions and queuing the email if one was thrown.

Just a suggestion.